### PR TITLE
fix: disallow unknown parameters in config

### LINKED ISSUE #3262: Disallow/warn about unknown parameters in config
Currently, if you put some unknown parameter in the config, Bee ignores it and does not "yell" or exit. While this saves us some work on Desktop for the upcoming 1.8 release (we don't have to write a migration to remove `transaction` and `block-hash` for existing configs), I would argue that this might not be the best behavior.

Mainly, because it creates space for user errors when the configuration of Bee, especially for typos.  For example, a user enters `ful-node: true` and wonders why Bee does not launch in Full mode even no errors seems to appear, but he has a typo `ful-node` instead of `full-node`.

I understand that this is a bit philosophical question about what should happen. You can go either with being strict and exit upon unknown parameters (honestly, my preference - fail fast), but if you would be against that, I would at least suggest putting a warning message to logs about the unknown parameters.

### PR DESCRIPTION
**Pull request by umerm-work**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
Closes #3262

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3782)

### CODE REVIEW

File: cmd/bee/cmd/deploy.go [hunk 1]
```diff
@@ -14,10 +14,13 @@ import (
 
 const blocktime = 15
 
+var missingParams []string
```
Comment (vladopajic) [inline cmd/bee/cmd/deploy.go:17]: does `missingParams` needs to be global variable? could it be local to `CheckMissingFlag` func?

**Review by vladopajic — APPROVED** [review]
LGMT

**Comment by vladopajic** [comment]
Heads up: There is similar PR currently open https://github.com/ethersphere/bee/pull/3272. @umerm-work, could you close #3272 once you merge this PR?

**Comment by umerm-work** [comment]
> There is similar PR currently

Yeah sure.

File: cmd/bee/cmd/start.go [hunk 2]
```diff
@@ -346,6 +347,22 @@ func buildBeeNode(ctx context.Context, c *command, cmd *cobra.Command, logger lo
 
 	return b, err
 }
+func (c *command) CheckMissingFlag(cmd *cobra.Command, args []string) error {
```
Comment (mrekucci) [inline cmd/bee/cmd/start.go:350]: This should be named `CheckUnknownParams` and it should be moved to the `cmd.go` file.

File: cmd/bee/cmd/start.go [hunk 3]
```diff
@@ -346,6 +347,22 @@ func buildBeeNode(ctx context.Context, c *command, cmd *cobra.Command, logger lo
...
+	if err := c.initConfig(); err != nil {
+		return err
+	}
+	var missingParams []string
```
Comment (mrekucci) [inline cmd/bee/cmd/start.go:354]: `missingParams` -> `unknownParams`

File: cmd/bee/cmd/start.go [hunk 4]
```diff
@@ -346,6 +347,22 @@ func buildBeeNode(ctx context.Context, c *command, cmd *cobra.Command, logger lo
...
+	for _, v := range c.config.AllKeys() {
+		if f := cmd.Flags().Lookup(v); f == nil {
+			missingParams = append(missingParams, v)
+		}
+	}
+	if len(missingParams) > 0 {
+		return errors.New("missing parameters " + strings.Join(missingParams, ","))
```
Comment (mrekucci) [inline cmd/bee/cmd/start.go:361]: Use `fmt.Errorf("unknown parameters:\n\t%v", strings.Join(missingParams, "\n\t"))` instead.

File: cmd/bee/cmd/start.go [hunk 5]
```diff
@@ -346,6 +347,22 @@ func buildBeeNode(ctx context.Context, c *command, cmd *cobra.Command, logger lo
 
 	return b, err
 }
+func (c *command) CheckMissingFlag(cmd *cobra.Command, args []string) error {
+	if err := c.initConfig(); err != nil {
+		return err
+	}
+	var missingParams []string
+	for _, v := range c.config.AllKeys() {
+		if f := cmd.Flags().Lookup(v); f == nil {
```
Comment (mrekucci) [inline cmd/bee/cmd/start.go:356]: Since you don't need the value of `f`, the following will do: `cmd.Flags().Lookup(v) == nil`.

**Review by mrekucci — APPROVED** [review]

**Pull request merged**